blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineNodeGraphVisitor.java

```diff
@@ -710,14 +711,14 @@ public BluePipelineStep getPipelineNodeStep(String id, Link parent) {
 
                 // New wrapper object based on current execution
                 FlowNodeWrapper newNodeWrapper = new FlowNodeWrapper(
-                    oldNodeWrapper.getNode(), // Use old node because we want to show the final id not intermediary
+                    currentNodeWrapper.getNode(),
                     currentNodeWrapper.getStatus(),
                     currentNodeWrapper.getTiming(),
                     currentNodeWrapper.getInputStep(),
                     run
                 );
 
-                newNodes.put(nodeId, newNodeWrapper);
+                newNodes.put(oldNodeWrapper.getId(), newNodeWrapper);
```

Comment:
Ok, so with some more testing, it turns out that this is actually the only change needed to fix the test case I am working with. The problem is that `nodeId` is an ID from the current build, but all other `get`s/`put`s on `newNodes` use an  ID from the old build, so when the IDs don't match across builds the logic in the rest this method can get mixed up. Regardless of the discussion in https://github.com/jenkinsci/blueocean-plugin/pull/2017#discussion_r316394955, I think we should use the ID from the old node here so that all of the keys in `newNodes` are IDs from the old build.